src/trace/step-function-service.ts

```diff
@@ -151,23 +151,23 @@ export class StepFunctionContextService {
   }
 
   private deterministicMd5HashInBinary(s: string): string {
-    // Md5 here is used here because we don't need a cryptographically secure hashing method but to generate the same trace/span ids as the backend does
-    const hex = Md5.hashStr(s);
+    const hash = new Sha256();
+    hash.update(s);
+    const hex = hash.digestSync().subarray(0, 16);
 
     let binary = "";
-    for (let i = 0; i < hex.length; i++) {
-      const ch = hex.charAt(i);
-      binary = binary + this.hexToBinary(ch);
+    for (const num of hex) {
+      binary = binary + this.numberToBinaryString(num);
     }
 
-    const res = "0" + binary.substring(1, 64);
-    if (res === "0".repeat(64)) {
+    const res = "0" + binary.substring(1, 128);
+    if (res === "0".repeat(128)) {
```

Comment:
Here, we need to ensure that the 0th and 64th bits of the 128-bit data are both set to 0.